### PR TITLE
Added create/read/update/delete for Item table.

### PR DESCRIPTION
**Pull request by himanshuparmar121**

### CODE REVIEW

File: Product.Service/ItemService.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿using Product.Data;
+using Product.Schema.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Product.Service
```
Comment (subhashgehlot) [inline Product.Service/ItemService.cs:9]: You should also have TenantId filteration here

File: Product.Service/ItemService.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                objItem.Name = item.Name;
+                objItem.UpdatedDate = DateTime.Now;
+                objItem.UnitOfMeasure = item.UnitOfMeasure; 
+            }
+
+            dbContext.SaveChanges();
+
+        }
+
+        public void RemoveItem(int id)
```
Comment (subhashgehlot) [inline Product.Service/ItemService.cs]: DeleteItem looks more appropriate

File: Product.Service/ItemService.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            Item? item = dbContext.Items.Where(x => x.Id == id).FirstOrDefault();
+
+            if (item != null)
+            {
+                dbContext.Items.Remove(item);
```
Comment (subhashgehlot) [inline Product.Service/ItemService.cs]: After remove do we require save changes?

File: Product.Service/ItemService.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            return dbContext.Items;
+        }
+
+        public void Update(Item item)
+        {
+            Item? objItem = dbContext.Items.Where(x => x.Id == item.Id).FirstOrDefault();
+
+            if (objItem != null)
+            {
+                objItem.Id = item.Id;
```
Comment (subhashgehlot) [inline Product.Service/ItemService.cs]: Where are child items here?
How are you going to update those, is that going to be through any other service?

File: Product.Service/ItemService.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace Product.Service
+{
+    public class ItemService 
+    {
+        private readonly ApplicationDbContext dbContext;
+
+        public ItemService(ApplicationDbContext _dbContext)
+        {
+            this.dbContext = _dbContext;
```
Comment (subhashgehlot) [inline Product.Service/ItemService.cs]: You should not use DbContext in service it should be only part of Data Project.
In service you will consume methods written in Data. You can use the same methods in Data project as you have done here

**Review by subhashgehlot — CHANGES_REQUESTED** [review]

File: Product.Service/ItemService.cs (hunk 1 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs:9]: How to add for TenantId like it shouldn't be null or any other filtration?

File: Product.Service/ItemService.cs (hunk 3 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs]: Can we do this after calling the remove or any other methods in the API itself?

File: Product.Service/ItemService.cs (hunk 2 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs]: Okay will do the same for all other methods as well.

File: Product.Service/ItemService.cs (hunk 1 above)
Comment (subhashgehlot) [reply Product.Service/ItemService.cs:9]: It should always contain TenantId in where clause.
Be it any query

File: Product.Service/ItemService.cs (hunk 3 above)
Comment (subhashgehlot) [reply Product.Service/ItemService.cs]: What is API here ?

File: Product.Service/ItemService.cs (hunk 1 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs:9]: okay

File: Product.Service/ItemService.cs (hunk 3 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs]: APIs that will be defined in the controller

Comment (subhashgehlot) [reply Product.Service/ItemService.cs]: So DbSave changes should happen in Data layer only

**Review by subhashgehlot — CHANGES_REQUESTED** [review]

File: Product.Service/ItemService.cs (hunk 3 above)
Comment (himanshuparmar121) [reply Product.Service/ItemService.cs]: okay

File: Product.Data/Item.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Product.Data
+{
+    public class Item
+    {
+        private readonly ApplicationDbContext dbContext;
```
Comment (subhashgehlot) [inline Product.Data/Item.cs]: Look for variable naming conventions

File: Product.Data/Item.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+        public Item(ApplicationDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        public void AddItem(Schema.Models.Item item)
```
Comment (subhashgehlot) [inline Product.Data/Item.cs:19]: Why you need to have fully qualified name here? Is there any conflict

File: Product.Data/Item.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            dbContext.Items.Add(item);
+            dbContext.SaveChanges();
+        }
+
+        public Schema.Models.Item? GetItemById(int? id, int? tenantId)
```
Comment (subhashgehlot) [inline Product.Data/Item.cs]: id and tenantId should be not null

File: Product.Data/Item.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+            if (objItem != null)
+            {
+                objItem.Id = item.Id;
+                objItem.TenantId = item.TenantId;
+                objItem.Name = item.Name;
+                objItem.UpdatedDate = DateTime.Now;
+                objItem.UnitOfMeasure = item.UnitOfMeasure;
+            }
+            dbContext.SaveChanges();
```
Comment (subhashgehlot) [inline Product.Data/Item.cs]: Take this inside if block.

File: Product.Data/Item.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        public void RemoveItem(int id, int tenantId)
```
Comment (subhashgehlot) [inline Product.Data/Item.cs:51]: Have one more method which will update just the isdelete flag instead of hard delete

**Review by subhashgehlot — CHANGES_REQUESTED** [review]

File: Product.Data/Item.cs (hunk 7 above)
Comment (himanshuparmar121) [reply Product.Data/Item.cs:19]: yes, as the Product.Data also contain same name 'Item' and 'ItemObjects' as Schema so I need to specify the full name.

File: Product.Data/Item.cs (hunk 8 above)
Comment (himanshuparmar121) [reply Product.Data/Item.cs]: Okay

**Review by subhashgehlot — APPROVED** [review]

**Review by subhashgehlot — APPROVED** [review]
Approved

**Pull request merged**